### ISSUE TITLE
Don't unselectAll when entering new edge mode

### ISSUE THREAD

**Issue by valtih1978** [almende/vis#612]
I have achieved this commenting `_unselectAll` in `createAddEdgeToolbar` and `if (this._getSelectedNodeCount() == 0) {` in `._handleConnect` like

```patch
@@ -32966,20 +32966,20 @@ return /******/ (function(modules) { // webpackBootstrap
    */
   exports._createAddEdgeToolbar = function() {
   console.log('_createAddEdgeToolbar')
     // clear the toolbar
     this._clearManipulatorBar();
-    this._unselectAll(true);
+    //this._unselectAll(true);
     this.freezeSimulation = true;
 
     if (this.boundFunction) {
       this.off('select', this.boundFunction);
     }
 
     var locale = this.constants.locales[this.constants.locale];
 
-    this._unselectAll();
+    //this._unselectAll();
     this.forceAppendSelection = false;
     this.blockConnectingEdgeSelection = true;
 
     this.manipulationDOM = {};
     this.manipulationDOM['backSpan'] = document.createElement('span');
@@ -33151,18 +33151,19 @@ return /******/ (function(modules) { // webpackBootstrap
    * to walk the user through the process.
    *
    * @private
    */
   exports._handleConnect = function(pointer) {
-    if (this._getSelectedNodeCount() == 0) {
+//    if (this._getSelectedNodeCount() == 0) {
       var node = this._getNodeAt(pointer);
 		console.log('handle connect ' + node)
       if (node != null) {
         if (node.clusterSize > 1) {
           alert(this.constants.locales[this.constants.locale]['createEdgeError'])
         }
         else {
+			this._unselectAll(true);
           this._selectObject(node,false);
           var supportNodes = this.sectors['support']['nodes'];
 
           // create a node the temporary line can look at
           supportNodes['targetNode'] = new Node({id:'targetNode'},{},{},this.constants);
@@ -33193,11 +33194,11 @@ return /******/ (function(modules) { // webpackBootstrap
 
           this.moving = true;
           this.start();
         }
       }
-    }
+  //  }
   };
 
   exports._finishConnect = function(event) {
 	console.log('finish connect ' + this._getSelectedNodeCount())
     if (this._getSelectedNodeCount() == 1) {

```

[Result looks](http://valtih1978.github.io/VisualDict/almende-vis.html) much more attractive/productive (control key for entering the mode).

**Comment by AlexDM0** [comment]
Hi,

Could you argue why it is bad that everything is unselected when you start the create new edge mode? Since you trigger it from a key shortcut I'm sure it is clearer but that was not the intended use. You don't have to share the few lines of code you altered to achieve this, if a discussion shows this behaviour is better than the one already in vis, we can change it easily.

Regards

**Comment by valtih1978** [comment]
Might be you right, this is unnecessary. I just tried to enter words into my graph and felt it more productive to keep selection. Newly added words are connected to currently selected ones, so that I do not need to waste time connecting them afterwards. I build a graph of related words, you know, and tend to enter related words together and it is nice to have them to stay selected even if I connect some of them additionally with control-mouse meanwhile. I see no point to remove selection. Extra code is evil even if it is not harmful. I have published the code for anybody who does not want the unnecessary unselect.
